quickwit/quickwit-search/src/root.rs

```diff
@@ -151,25 +151,40 @@ pub struct IndexMetasForLeafSearch {
 }
 
 pub(crate) type IndexesMetasForLeafSearch = HashMap<IndexUid, IndexMetasForLeafSearch>;
-type TimestampFieldOpt = Option<String>;
+
+#[derive(Debug)]
+struct RequestMetadata {
+    timestamp_field_opt: Option<String>,
+    query_ast_resolved: QueryAst,
+    indexes_meta_for_leaf_search: IndexesMetasForLeafSearch,
+    sort_fields_are_datetime: Vec<bool>,
```

Comment:
`sort_field_ordinal_is_datetime`? Or better would be a Hashmap as long it's not performance critical